client/etcd/client.go

```diff
@@ -124,24 +133,36 @@ func (c *csclient) createServices() {
 	})
 }
 
-func (c *csclient) createTxnStore() {
-	c.kvOnce.Do(func() {
-		opts := etcdkv.NewOptions().
-			SetInstrumentsOptions(instrument.NewOptions().
-				SetLogger(c.logger).
-				SetMetricsScope(c.kvScope)).
-			SetPrefix(prefix(c.opts.Env()))
-		c.kv, c.kvErr = c.txnGen(opts, c.opts.Zone())
+func (c *csclient) createTxnStore(namespace string) (kv.TxnStore, error) {
+	c.txnOnce.Do(func() {
+		c.txn, c.txnErr = c.txnGen(c.opts.Zone(), namespace, c.opts.Env())
 	})
+	return c.txn, c.txnErr
 }
 
-func (c *csclient) kvGen(kvOpts etcdkv.Options) sdclient.KVGen {
+func (c *csclient) kvGen() sdclient.KVGen {
 	return sdclient.KVGen(func(zone string) (kv.Store, error) {
-		return c.txnGen(kvOpts, zone)
+		return c.txnGen(zone)
 	})
 }
 
-func (c *csclient) txnGen(kvOpts etcdkv.Options, zone string) (kv.TxnStore, error) {
+func (c *csclient) newkvOptions(zone string, namespaces ...string) etcdkv.Options {
+	opts := etcdkv.NewOptions().
+		SetInstrumentsOptions(instrument.NewOptions().
+			SetLogger(c.logger).
+			SetMetricsScope(c.kvScope)).
+		SetCacheFileFn(c.cacheFileFn(zone))
+
+	for _, namespace := range namespaces {
+		if namespace == "" {
+			continue
+		}
+		opts = opts.SetPrefix(opts.ApplyPrefix(namespace))
```

Comment:
no, it will be in the same order as the namespaces passed in, pretty much just keep appending namespaces